### ISSUE TITLE
Able to set --cluster-dns

### ISSUE THREAD

**Issue by mzupan** [bottlerocket-os/bottlerocket#2132]
**What I'd like:**

Being able to set --cluster-dns flag for kubelet. This allows the kubelet to set the dns search to something other then coredns like node-local-dns then fail back to coredns

**Comment by gthao313** [comment]
Hi @mzupan  Thanks for opening this issue. Bottlerocket team will research on this and reach back to you soon. Thanks.

**Comment by mzupan** [comment]
@gthao313 thanks how it was used with just a linux ami in eks was like

```
 --cluster-dns=169.254.20.10,172.20.0.10
```

The first IP being the IP node-local sets on each node and the 2nd being coredns.

That set any pod running on that node to have a resolv.conf that looked like

```
❯ k exec -it debug-6ff4966bc5-dn6tq -- cat /etc/resolv.conf
nameserver 169.254.20.10
nameserver 172.20.0.10
search default.svc.cluster.local svc.cluster.local cluster.local ec2.internal
options ndots:5
```

**Comment by bcressey** [comment]
Does `settings.kubernetes.cluster-dns-ip` not work for this? We generate a value for the setting automatically, but if you specify a value that should override the default.

**Comment by mzupan** [comment]
@bcressey that doesn't seem to allow you to specify multiple values like `--cluster-dns` in the kubelet does. Really there for if node-local somehow fails it uses coredns as a failback

**Comment by bcressey** [comment]
Now that you point it out, it's obvious that it can be a list of IPs in the [kubelet config template](https://github.com/bottlerocket-os/bottlerocket/blob/6a82bc56b203ab27c78d5d043c3b4f4d5f1a68c8/packages/kubernetes-1.22/kubelet-config#L31), but it never crossed my mind to make it a list in our API. 😄 

@etungsten seems like we need some of the serde magic here to migrate `cluster-dns-ip` into a list shape on update.

**Comment by cbgbt** [comment]
This is resolved in the latest Bottlerocket release, [v1.8.0](https://github.com/bottlerocket-os/bottlerocket/releases/tag/v1.8.0) -- `settings.kubernetes.cluster-dns-ip` now supports being set to an array of IP addresses. Check [the documentation](https://github.com/bottlerocket-os/bottlerocket#kubernetes-settings) for an example on setting this.

Let us know if you have any questions!
